pkg/controller/deployment/sync.go

```diff
@@ -247,14 +303,38 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 	newRevision := strconv.FormatInt(maxOldRevision+1, 10)
 
 	existingNewRS, err := deploymentutil.FindNewReplicaSet(deployment, rsList)
-	if err != nil {
+	switch {
+	case err != nil:
 		return nil, err
-	} else if existingNewRS != nil {
-		// Set existing new replica set's annotation
+
+	case existingNewRS != nil:
+		// Latest replica set exists. We need to sync its annotations (includes copying all but
+		// annotationsToSkip from the parent deployment, and update revision, desiredReplicas,
+		// and maxReplicas) and also update the revision annotation in the deployment with the
+		// latest revision.
 		if deploymentutil.SetNewReplicaSetAnnotations(deployment, existingNewRS, newRevision, true) {
-			return dc.client.Extensions().ReplicaSets(deployment.ObjectMeta.Namespace).Update(existingNewRS)
+			if _, err := dc.client.Extensions().ReplicaSets(deployment.ObjectMeta.Namespace).Update(existingNewRS); err != nil {
```

Comment:
yes but that's fine since in the next iteration the replica set will not be there